pkg/authn/config.go

```diff
@@ -0,0 +1,24 @@
+package authn
+
+type AuthnHeaderConfig struct {
+	// When set to true, kube-rbac-proxy adds auth-related fields to the headers of http requests sent to the upstream
+	Enabled bool
+	// Corresponds to the name of the field inside a http(2) request header
+	// to tell the upstream server about the user's name
+	UserFieldName string
+	// Corresponds to the name of the field inside a http(2) request header
+	// to tell the upstream server about the user's groups
+	GroupsFieldName string
+	// The separator string used for concatenating multiple group names in a groups header field's value
+	GroupSeparator string
+}
+
+type AuthnConfig struct {
```

Comment:
Same as above. Also, we should clarify the difference of `AuthnConfig` and `AuthnHeaderConfig` in the godoc.